services/auth.js

```diff
@@ -66,7 +66,11 @@ const authService = {
       throw createError(401, INCORRECT_CREDENTIALS)
     }
 
-    const tokens = tokenService.generateTokens({ id: user._id, role: user.role.value })
+    if (user.isFirstLogin) {
+      await User.updateOne({ _id: user._id }, { $set: { isFirstLogin: false } }).exec()
```

Comment:
here you set isFirstLogin to false on first login, because isFirstLogin is true by default

Reply:
we need to do this after we send response

Reply:
I just noticed my mistake, but I think we need to set it just after the token generation, because after it, it doesn't matter, so if you're okay with that, I'll fix it to that

Reply:
yep